### PR TITLE
Bifrost CLI/Shell

### PR DESCRIPTION
**Pull request by SeanCheatham**
## Purpose
- To participate in staking & block production, a user must generate several secret keys and a Transaction which gets included in the chain
## Approach
- Create a CLI/shell that guides a user through the key generation and transaction creation process
## Testing
- image
## Tickets
- #BN-908
## Usage
- `sbt "node/run cli"`
## Notes
- Follow-up work is still needed to load these keys + transaction from disk for _public_ networks

### CODE REVIEW

File: node/src/main/scala/co/topl/node/cli/CliApp.scala [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+  /**
+   * Generate and save a VRF key.
+   */
+  private val createVrfKey =
+    StageResultT.liftF[F, Unit](
+      c.println("Generating a VRF Key.  This key establishes your \"randomness\" within the blockchain.")
+    ) >>
+    createSeed
+      .map(Ed25519VRF.precomputed().deriveKeyPairFromSeed)
+      .flatTap(key => write(key._1)("VRF SK", "operator-key.ed25519.sk"))
```
Comment (nandotorterolo) [inline node/src/main/scala/co/topl/node/cli/CliApp.scala:200]: question, at the end of the process, only 1file should be created?
example the operator key will be overridden by the vrf key?

File: node/src/main/scala/co/topl/node/cli/CliApp.scala [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+      Sync[F]
+        .delay(EntropyToSeed.instances.pbkdf2Sha512(32).toSeed(Entropy.generate(), password = None))
+    )
+
+  /**
+   * Generate and save an Ed25519 Operator key.
+   */
+  private val createOperatorKey =
+    StageResultT.liftF[F, Unit](
+      c.println("Generating an Ed25519 Operator SK.  This key determines your StakingAddress.")
```
Comment (nandotorterolo) [inline node/src/main/scala/co/topl/node/cli/CliApp.scala:185]: Does this require support on Ed25519 on brambl side?

**Review by nandotorterolo — APPROVED** [review]
Looks good, great PR, just a couple of questions.

File: node/src/main/scala/co/topl/node/cli/CliApp.scala (hunk 1 above)
Comment (SeanCheatham) [reply node/src/main/scala/co/topl/node/cli/CliApp.scala]: Good catch, thanks!  This should have been a separate file.  I updated the PR.

File: node/src/main/scala/co/topl/node/cli/CliApp.scala (hunk 2 above)
Comment (SeanCheatham) [reply node/src/main/scala/co/topl/node/cli/CliApp.scala:196]: The "Operator Key" isn't used for Locks on UTxOs; at this time it is only used to prove/demonstrate that you own a particular StakingAddress

**Pull request merged**